### PR TITLE
[SD-3031] Removal of images when expiring from ingest

### PR DESCRIPTION
**Pull request by marwoodandrew**
It seems that the clean images routine was broken.

Also the images are not getting purged as the cursor had been traversed already when attempting to construct the list of file id's

### CODE REVIEW

**Comment by sivakuna-aap** [comment]
:+1:

**Comment by akintolga** [comment]
:+1:

File: server/superdesk/io/commands/update_ingest_tests.py [hunk 1]
```diff
@@ -225,6 +225,26 @@ def test_query_getting_expired_content(self):
...
+            remove = RemoveExpiredContent()
+            remove.run(provider.get('type'))
+
+            # only one left in ingest
+            after = service.get(req=None, lookup={})
+            self.assertEqual(1, after.count())
+
+            # all gone
+            current_files = app.media.fs('upload').find()
+            self.assertEqual(0, current_files.count())
```
Comment (plamut) [inline server/superdesk/io/commands/update_ingest_tests.py]: In general each unit test case should only test a single use case (i.e. one execution path), resulting in more fine-grained failure messages, pinpointing the exact execution path that doesn't work. I would thus suggest splitting this test into four separate test cases.

In general it's good though.

**Comment by plamut** [comment]
:+1:

**Pull request merged**